File: routes/menu.ts
// Menu Routes
import { Express } from "express";

export function registerMenuRoutes(app: Express, db: any) {
  app.get("/api/menu/today", (req, res) => {
    try {
      const today = new Date().toISOString().split('T')[0];
      const menu = db.prepare("SELECT * FROM daily_menu WHERE date = ?").all(today);
      if (menu.length === 0) {
        res.json([
          { meal_type: 'breakfast', items: ['Poha', 'Jalebi', 'Tea'], status: 'completed' },
          { meal_type: 'lunch', items: ['Rice', 'Dal Tadka', 'Paneer Masala', 'Roti', 'Salad'], status: 'ongoing' },
          { meal_type: 'dinner', items: ['Jeera Rice', 'Chole', 'Roti', 'Gulab Jamun'], status: 'upcoming' }
        ]);
      } else {
        res.json(menu.map((m: any) => ({
          ...m, items: m.food_item_ids ? JSON.parse(m.food_item_ids) : []
        })));
      }
    } catch (err: any) {
      res.status(500).json({ success: false, error: err.message });
    }
  });

  app.get("/api/menu/week", (req, res) => {
    try {
      const menus = db.prepare("SELECT * FROM daily_menu").all();
      res.json({ success: true, data: menus });
    } catch (err: any) {
      res.status(500).json({ success: false, error: err.message });
    }
  });

  app.post("/api/menu", (req, res) => {
    try {
      const { date, meal_type, items } = req.body;
      db.prepare("INSERT OR REPLACE INTO menu (date, meal_type, food_item_ids) VALUES (?, ?, ?)").run(date, meal_type, JSON.stringify(items));
      res.json({ success: true });
    } catch (err: any) {
      res.status(500).json({ success: false, error: err.message });
    }
  });
}

File: backend/routes/menu.py
from fastapi import APIRouter
from pydantic import BaseModel
from typing import List
from datetime import datetime
from database import db
import json

router = APIRouter(prefix="/api/menu", tags=["menu"])


class MenuCreate(BaseModel):
    date: str
    meal_type: str
    items: List[str]


@router.get("/today")
def today_menu():
    today = datetime.now().strftime("%Y-%m-%d")
    menu = db.filter("daily_menu", date=today)
    if not menu:
        return [
            {"meal_type": "breakfast", "items": ["Poha", "Jalebi", "Tea"], "status": "completed"},
            {"meal_type": "lunch", "items": ["Rice", "Dal Tadka", "Paneer Masala", "Roti", "Salad"], "status": "ongoing"},
            {"meal_type": "dinner", "items": ["Jeera Rice", "Chole", "Roti", "Gulab Jamun"], "status": "upcoming"},
        ]
    result = []
    for m in menu:
        items = m.get("food_item_ids", "[]")
        if isinstance(items, str):
            try:
                items = json.loads(items)
            except Exception:
                items = []
        result.append({**m, "items": items})
    return result


@router.get("/week")
def week_menu():
    return {"success": True, "data": db.get_all("daily_menu")}


@router.post("")
def create_menu(menu: MenuCreate):
    existing = [m for m in db.get_all("daily_menu") if m.get("date") == menu.date and m.get("meal_type") == menu.meal_type]
    if existing:
        db.update("daily_menu", existing[0]["id"], {"food_item_ids": json.dumps(menu.items)})
    else:
        db.insert("daily_menu", {"date": menu.date, "meal_type": menu.meal_type, "food_item_ids": json.dumps(menu.items)})
    return {"success": True}
